ehrapy/api/_anndata_util.py

```diff
@@ -36,3 +36,73 @@ def get_column_values(adata: AnnData, indices: Union[int, List[int]]) -> np.ndar
         :class:`~numpy.ndarray` object containing the column values
     """
     return np.take(adata.X, indices, axis=1)
+
+
+def assert_encoded(adata: AnnData):
+    try:
+        assert "categoricals" in adata.uns_keys()
+    except AssertionError:
+        raise NotEncodedError("The AnnData object has not yet been encoded.") from AssertionError
+
+
+def get_numeric_vars(adata: AnnData) -> List[str]:
+    """Fetches the column names for numeric variables in X.
+
+    Args:
+        adata: :class:`~anndata.AnnData` object
+
+    Returns:
+        Set of column numeric column names
+    """
+
+    assert_encoded(adata)
+
+    return adata.uns["categoricals"]["not_categorical"]
```

Comment:
@Zethson , @lazappi  ; Didn't test it myself but I guess we may run into issues here, considering the following scenario:

Encoding happens via `autodetect`. When this is the case we have three "types" of categories: The `non_categoricals` (used here), the `categoricals` that needs to be encoded and a third group that is stored under `categoricals`, but does not need any encodings since it's already numerical. 

I don't remember why we made a difference in the very beginning between those two types of categoricals (do you @Zethson ?). Maybe its completely unnecessary and we can remove it? Because those are the numerical values, that won't be returned here.

Reply:
"a third group that is stored under categoricals, but does not need any encodings since it's already numerical."

Maybe this was for columns like e.g sex? sex in the MIMIC2-IAC dataset is 0 or 1. Categorical but numerical.

Reply:
Yes, but where are we actually using this? Do we need this?

Reply:
No not really. Should not be an issue with coloring. Thing is, I had the impression that we had this well mapped out so there must be some use-case for this that I cannot remember.
I'll need a fresh brain for this... 
If we don't come up with better ideas we can remove this ofc.

But atm not an issue for @lazappi right? He can just make the assumption that it does not matter for now.

Reply:
I mean, if those "numerical" categoricals should not be normalized in any way, then it's no issue.

Reply:
Yeah I noticed that an integer column I had was put into the "categorical but not encoded" category. I meant to ask about that but just assumed you chose to have it that way for some reason.

Reply:
Aye. See my comment in the whole PR. Everything is fine :)